CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -150,28 +146,51 @@ public class Child: NodeChoiceConvertible {
   /// For any other kind of child nodes, accessing this property crashes.
   public var syntaxChoicesType: TypeSyntax {
     precondition(kind.isNodeChoices, "Cannot get `syntaxChoicesType` for node that doesn’t have nodeChoices")
-    return "\(raw: name.withFirstCharacterUppercased)"
+    return "\(raw: newestName.withFirstCharacterUppercased)"
   }
 
   /// If this child only has tokens, the type that the generated `TokenSpecSet` should get.
   ///
   /// For any other kind of child nodes, accessing this property crashes.
   public var tokenSpecSetType: TypeSyntax {
     precondition(kind.isToken, "Cannot get `tokenSpecSetType` for node that isn’t a token")
-    return "\(raw: name.withFirstCharacterUppercased)Options"
+    return "\(raw: newestName.withFirstCharacterUppercased)Options"
   }
 
-  /// The deprecated name of this child that's suitable to be used for variable or enum case names.
-  public var deprecatedVarName: TokenSyntax? {
-    guard let deprecatedName = deprecatedName else {
-      return nil
+  /// If this child is actually part of another child's history, links back
+  /// to that child (and its parent, if this child was extracted to another
+  /// node). Empty if this is the newest version of the child.
```

Comment:
I had a hard time grasping this (I think because I thought that the array traced the name across multiple change sets). I think an example would go a long way here to illustrate when `newerChildPath` can have multiple entries.